### PR TITLE
Adding go vet to test makefile target

### PR DESCRIPTION
**Pull request by feclare**
Also fixing the issues reported.

### CODE REVIEW

File: core/clients/metadataclients/metadata-clients.go [hunk 1]
```diff
@@ -193,7 +194,7 @@ func (a *AddressableClient) AddressableForName(name string) (models.Addressable,
 	}
 	defer resp.Body.Close()
 	if err != nil {
-		fmt.Println("AddressableForName makeRequest failed: %s", err)
+		fmt.Printf("AddressableForName makeRequest failed: %v\n", err)
```
Comment (drasko) [inline core/clients/metadataclients/metadata-clients.go]: Do we need hese printf lines? I would like to remove all of these in the production and use only logs.

Comment (feclare) [reply core/clients/metadataclients/metadata-clients.go]: I agree with you. But in this case I just wanted to remove the vet warnings.

We will need to cleanup all the codebase logs.

Comment (drasko) [reply core/clients/metadataclients/metadata-clients.go]: Agreed on this.

**Comment by drasko** [comment]
LGTM

**Pull request merged**